git_machete/cmd.py

```diff
@@ -1676,6 +1619,90 @@ def slurp_file_or_empty(path: str) -> str:
         self.__annotations[head] = f'PR #{pr.number}'
         self.save_definition_file()
 
+    def __handle_diverged_and_newer_state(self, cb: str, remote: str, abort: bool = True) -> None:
+        self.__print_new_line(False)
+        rb = self.__git.strict_counterpart_for_fetching_of_branch(cb)
+        ans = self.ask_if(
+            f"Branch {bold(cb)} diverged from (and has newer commits than) its remote counterpart {bold(rb)}.\n"
+            f"Push {bold(cb)} with force-with-lease to {bold(remote)}?" + pretty_choices('y', 'N', 'q', 'yq'),
+            f"Branch {bold(cb)} diverged from (and has newer commits than) its remote counterpart {bold(rb)}.\n"
+            f"Pushing {bold(cb)} with force-with-lease to {bold(remote)}...",
+            override_answer=None if self.__cli_opts.opt_push_tracked else "N")
+        if ans in ('y', 'yes', 'yq'):
+            self.__git.push(remote, cb, force_with_lease=True)
+            if ans == 'yq':
+                if not abort:
+                    return
+                raise MacheteException('Process interrupted.')
```

Comment:
Hmm I think displaying such a message to the user might sound a bit too... _stiff_? technical?

Reply:
Consider using `StopTraversal` exception, it's been created for the purpose of returning a few methods when quitting `traverse`

Reply:
I agree, this is too technical, but on the other hand, would be perfect if the message here fits both contexts: `traverse` and `__sync_before_creating_pr`.  
BTW don't you think that wee need specialized exception here as issue #164 says?

Reply:
> I agree, this is too technical, but on the other hand, would be perfect if the message here fits both contexts: traverse and __sync_before_creating_pr.

Okay, but how about just displaying NO message? I think that would also fit both contexts 🤔

Reply:
> BTW don't you think that wee need specialized exception here as issue #164 says?

As in, an exception different than `StopTraversal`, but with the same effect on the execution? 🤔